detect_secrets/core/usage/scan.py

```diff
@@ -7,7 +7,14 @@
 
 
 def add_scan_action(parent: argparse._SubParsersAction) -> argparse.ArgumentParser:
-    parser = parent.add_parser('scan')
+    parser = parent.add_parser(
+        'scan',
+        help='Creates a baseline by scanning a repository for secrets.',
+        description=(
+            'Scans a repository for secrets in code. The generated output is compatible with '
+            '`detect-secrets-hook --baseline`.'
```

Comment:
An aside: We have had a lot of users, I can pull up references, that want `detect-secrets` itself to take a baseline as well. Lots of people don't use pre-commit hooks, they might use arc linters and other things along those lines, and want to use `--baseline` argument without all the same assumptions that went into the pre-commit hook.

Curious what your thoughts are on this.

Reply:
I think you may need to elaborate on this, but v1 does intend to support this. Essentially, the breakdown of the different usages of this tool include:

- `detect-secrets scan`: scans secrets, and adds new secrets to a baseline
- `detect-secrets-hook`: blocks new secrets
- `detect-secrets audit`: analyzes results

In v1, scan does take in a baseline (e.g. `detect-secrets scan --baseline .secrets.baseline`), but since scan's job is to create a baseline, it uses this baseline as a configuration file, and updates it with the latest secrets. The only assumptions in the pre-commit hook (as written) is:

- new secrets should be blocked
- people should be able to see which new secrets were blocked
- baselines should be regularly updated (which can be reduced with a WIP PR for removal of line numbers)

The hook should not be dependent on the `pre-commit` framework, which I intend to address via better documentation.